### ISSUE TITLE
HMA throws exception with a misleading / incorrect message.

### ISSUE THREAD

**Issue by bobtabor** [DaveSkender/Stock.Indicators#518]
## The problem

I called:

```
var values = quotes.GetHma(9);
```
Where `quotes` only had 10 items.  This throws an exception, which is fine.  The **text** of the exception is not quite right.  (See next section.)

Error message(s):

```
Skender.Stock.Indicators.BadQuotesException: 'Insufficient quotes provided for WMA.  You provided 2 periods of quotes when at least 3 is required. (Parameter 'quotes')'
```

## To Reproduce

Call the `GetHma()` extension method passing in `9` as the number of lookback periods, calling from a List<Quote> with 10 instances of `Quote`.

## Expected behavior

I expected the error message use the term "HMA" rather than "WMA", and I expected the count of items to reflect the actual number of items in my collection (which was 10).

## Screenshots or other reference materials

![exception](https://user-images.githubusercontent.com/45040176/129104698-636df24c-f30d-46d2-a306-3893a846bff4.PNG)

**Comment by DaveSkender** [comment]
I'm glad you're enjoying it!  Thank you.

And thanks for finding this interestingly strange bug.  HMA uses WMA internally and on their face, they have the same historical quotes requirements.  HMA basically takes two WMAs then does another aggregate calculation that uses the square root of `N` quotes.  Square root of 9, rounded is 3, and then it uses `N-1` quotes.

Clearly a rounding error in here that pops up with small histories.  I may also have to increase the minimum threshold if that is the actual problem.  I'll dig in on it this weekend.

**Comment by DaveSkender** [comment]
Turns out this has less to do with the smallness of history, but rather the calculation of minimum history.  Originally, I was saying you need a minimum of `N` periods of historical quotes, where `N` is the lookback period.  The math on this is hard to follow in the formula, but it turns out the calculation needs a minimum of `N+(integer of SQRT(N))-1` quotes.

**Comment by DaveSkender** [comment]
This was fixed in version 1.15.1
